validation/cubed_sphere_eddying_aquaplanet/cubed_sphere_eddying_aquaplanet.jl

```diff
@@ -216,15 +216,12 @@ function cubed_sphere_eddying_aquaplanet(grid_filepath)
     deformation_radius_45°N = √(g * H) / (2Ω*sind(45))
     @info @sprintf("Deformation radius @ 45°N: %.2f km", deformation_radius_45°N / 1000)
 
-    cfl = CFL(Δt, accurate_cell_advection_timescale)
+    # cfl = CFL(Δt, accurate_cell_advection_timescale)
+    cfl = 0.2
 
-    simulation = Simulation(model,
-                        Δt = Δt,
-                 stop_time = 5years,
-        iteration_interval = 20,
-                  progress = Progress(time_ns()),
-                parameters = (; cfl)
-    )
+    simulation = Simulation( model, Δt=Δt, stop_time=5years)
+    # wizard = TimeStepWizard(cfl=cfl)
+    # simulation.callbacks[:wizard] = Callback(wizard, IterationInterval(20))
```

Comment:
You might want

```julia
simulation.callbacks[:progress] = Callback(Progress(time_ns()), IterationInterval(20))
```

if you want that progress printing.

`parameters` is old indeed.

Reply:
Hmmmm - tried :progress but got 
```
[2022/01/21 11:44:19.367] INFO  Gravity wave CFL = 0.2279
[2022/01/21 11:44:19.370] INFO  Deformation radius @ 45°N: 68.57 km
[2022/01/21 11:46:01.174] INFO  Initializing simulation...
ERROR: type Simulation has no field run_time
Stacktrace:
  [1] getproperty
    @ ./Base.jl:42 [inlined]
  [2] (::Progress)(sim::Simulation{HydrostaticFreeSurfaceModel{Oceananigans.TimeSteppers.QuasiAdamsBashforth2TimeStepper{Float64, NamedTuple{(:u, :v, :η), Tuple{Field{Face, Center, Center, Nothing, ConformalCubedSphereGrid{Float64, NTuple{6, ConformalCubedSphereFaceGrid{Float64, Periodic, Periodic, Bounded, OffsetArrays.OffsetMatrix{Float64, Matrix{Float64}}, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}}, CPU}}, NTuple{6, Oceananigans.CubedSpheres.CubedSphereFaceConnectivity{Oceananigans.CubedSpheres.CubedSphereFaceConnectivityDetails{Int64, Symbol}, Oceananigans.CubedSpheres.CubedSphereFaceConnectivityDetails{Int64, Symbol}, Oceananigans.CubedSpheres.CubedSphereFaceC
......
```

Reply:
dont know where `Progress` is defined, but that property is now called `run_wall_time`

Reply:
can write `propertynames(simulation)` to see a list at the REPL

Reply:
Hmm that's weird there is a `run_wall_time`

```
julia> simulation = cubed_sphere_eddying_aquaplanet(datadep"cubed_sphere_96_grid/cubed_sphere_96_grid.jld2")
[2022/01/21 12:13:17.174] WARN  ConformalCubedSphereGrid is experimental: use with caution! -@-> /Users/chrishill/projects/onan-2022-01-20/src/CubedSpheres/conformal_cubed_sphere_grid.jl:161
[2022/01/21 12:13:22.393] INFO  Gravity wave CFL = 0.2279
[2022/01/21 12:13:22.394] INFO  Deformation radius @ 45°N: 68.57 km
Simulation{typename(HydrostaticFreeSurfaceModel){typename(CPU), Float64}}
├── Model clock: time = 0 seconds, iteration = 0
├── Next time step: 10 minutes
├── Elapsed wall time: 0 seconds
├── Stop time: 5 years
├── Stop iteration : Inf
├── Wall time limit: Inf
├── Callbacks: typename(OrderedCollections.OrderedDict) with 5 entries:
│   ├── stop_time_exceeded => typename(Callback)
│   ├── stop_iteration_exceeded => typename(Callback)
│   ├── wall_time_limit_exceeded => typename(Callback)
│   ├── nan_checker => typename(Callback)
│   └── progress => typename(Callback)
├── Output writers: typename(OrderedCollections.OrderedDict) with 1 entry:
│   └── fields => typename(JLD2OutputWriter)
└── Diagnostics: typename(OrderedCollections.OrderedDict) with no entries

julia> propertynames(simulation)
(:model, :timestepper, :Δt, :stop_iteration, :stop_time, :wall_time_limit, :diagnostics, :output_writers, :callbacks, :run_wall_time, :running, :initialized)

julia> 
```

Reply:
Grrrrr 🥲 

```
julia> run_cubed_sphere_eddying_aquaplanet()
[2022/01/21 12:22:15.488] WARN  ConformalCubedSphereGrid is experimental: use with caution! -@-> /Users/chrishill/projects/onan-2022-01-20/src/CubedSpheres/conformal_cubed_sphere_grid.jl:161
[2022/01/21 12:24:12.342] INFO  Gravity wave CFL = 0.2279
[2022/01/21 12:24:12.344] INFO  Deformation radius @ 45°N: 68.57 km
[2022/01/21 12:25:45.160] INFO  Initializing simulation...
ERROR: InexactError: trunc(Int64, NaN)
Stacktrace:
  [1] trunc
    @ ./float.jl:805 [inlined]
  [2] round
    @ ./float.jl:369 [inlined]
  [3] (::Progress)(sim::Simulation{HydrostaticFreeSurfaceModel{Oceananigans.TimeSteppers.QuasiAdamsBashforth2TimeStepper{Float64, NamedTuple{(:u, :v, :η), Tuple{Field{Face, Center, Center, Nothing, ConformalCubedSphereGrid{Float64, NTuple{6, ConformalCubedSphereFaceGrid{Float64, Periodic, Periodic, Bo
```

Reply:
Might be because `run_wall_time` is 0 at iteration 0 or something.

Reply:
You don't really need an ETA calculation anyways 😏

Reply:
iteration_interval RIP too! a short, sweet life..... 💀 ✝️ 

```
Initializing simulation...
hello progress
progress = 0.0
sim.run_wall_time = 0.0
sim.model.clock.time = 0.0
sim.stop_time = 1.5768e8
goodbye progress
[2022/01/21 14:07:45.604] INFO  [000.00%] Time: 0 seconds, iteration: 0, max(|u⃗|): (4.29e-01, 4.28e-01) m/s, extrema(η): (min=0.00e+00, max=0.00e+00)
ERROR: type Simulation has no field iteration_interval
Stacktrace:
  [1] getproperty
    @ ./Base.jl:42 [inlined]
  [2] macro expansion
    @ ./logging.jl:360 [inlined]
  [3] (::Progress)(sim::Simulation{HydrostaticFreeSurfaceModel{Oceananigans.TimeSteppers.QuasiAdamsBashforth2TimeStepper{Float64, NamedTuple{(:u, :v, :η), Tuple{Field{Face, Center, Center, Nothing, ConformalCubedSphereGrid{Float64, NTuple{6, ConformalCubedSphereFaceGrid{Float64, Periodic, Periodic, Bounded, OffsetArrays.OffsetMatrix{Float64, Matrix{Float64}}, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision
```

Reply:
We have a much more powerful interface for callbacks now! Every callback has their own `schedule`, which can be `IterationInterval`...